### ISSUE TITLE
Can't recover Electrum 1.x seeds using Seedlist or Tokenlist, Electrum 1.x seeds don't work with AddressDB Recovery

### ISSUE THREAD

**Issue by csavas03** [3rdIteration/btcrecover#55]
When trying to recover Electrum 1.x seeds from either a seedlist or tokenlist, I get this error:
`WalletElectrum1 does not support the lang option`
If I remove the `--language en` option, I get this error instead:
`Error: Language needs to be specificed if using tokenlist or passwordlist`
Here is the command I was using:
`$ python seedrecover.py --dsw --wallet-type electrum1 --addressdb addresses.db --addr-limit 1 --seedlist seeds.txt --mnemonic-length 12 --language en`
If I replace "electrum1" with "electrum2", it seems to work.

**Comment by 3rdIteration** [comment]
Should be fixed in https://github.com/3rdIteration/btcrecover/commit/a8b00ea0272fce1c9cb973329e12aeed7a901bdb

Just be aware that Electrum 1 seed checking is thousands of times slower than Electrum2 and BIP39 seeds, so completely descrambling a seed will take a few months on a typical PC. (Though it would probably benefit from GPU acceleration which wouldn't be difficult to extend to this wallet type)

That said, if you know the location of some words and there are only tens of millions of possibilities, it is totally workable in a few days.

**Comment by csavas03** [comment]
I tried running the exact same command as before, but it hangs on:
`Phase 1/1: up to 12 mistakes, 12 of which can be an entirely different seed word.` after loading the addressdb.
I even made a new seedlist file with only a single line, but I got the same result. (It prints the seedlist correctly with `--listseeds` though)

**Comment by 3rdIteration** [comment]
Interesting, I just had a dig and it turns out that Electrum1 seed recovery probably hasn't worked with an AddressDB for a few years now, as it was using a really inefficient way to check against the list of addresses. (Which would take hours with a Bitcoin sized AddressDB)

Anyway it has been merged now in https://github.com/3rdIteration/btcrecover/commit/0102dfb47556f54d77e13c2886b68dd3bd76483b so both tokenlist recovery for Electrum1 seeds and AddressDB recovery should work fine.

**Comment by giantmonster** [comment]
Unfortunately, I'm having some trouble with this code. 
When I use the --no-dupchecks --no-eta flags, it's giving an error of 
<ValueError: tuple.index(x): x not in tuple>
When I don't use these it kicks out after running through initial set-up with the following error message.
<at least 30,800,000 passwords to try, ETA > --max-eta option (168 hours), exiting> 
Thanks for your efforts!

**Comment by 3rdIteration** [comment]
Just in terms of reproduction on my end, it's 30 million the length of your seedlist?

**Comment by giantmonster** [comment]
I'm using a token-list that I assume is generating that number with error checking changing possible outcomes. I don't mind running the process for months if needed. I can, however, reduce as your question suggests.

**Comment by 3rdIteration** [comment]
I just tried running this and can't reproduce the error with the --no-eta flag. (The nodupchecks one is automatically applied to this kind of test, you can you leave that off.

Can you try running this command:

python seedrecover.py --dsw --wallet-type electrum1 --addressdb E:\Crypto\addresses-BTC.db --tokenlist d:\SeedTokenList_ElectrumV1.txt --mnemonic-length 12 --language en --addr-limit 5 --no-eta

The address it is looking for is 12zAz6pAB6LhzGSZFCc6g9uBSWzwESEsPT which I only just added a TX on (so it won't appear as part of your address DB unless you manually add it) but I'm interested to see if it at least starts running.

With this example tokenlist:
[SeedTokenList_ElectrumV1.txt](https://github.com/3rdIteration/btcrecover/files/5256843/SeedTokenList_ElectrumV1.txt)

**Comment by giantmonster** [comment]
Hi! Your example is processing using the SeedTokenList_ElectrumV1.txt in the test/test-listfiles/ directory, however without a successful outcome. I'll try with the file you have in the above message.
Also, I'll match my token file syntax to what you have in the SeedTokenList_ElectrumV1.txt file. I have more words however and additional three encryption words.

**Comment by giantmonster** [comment]
I'm using the command line you provided above. The code is working however, the search takes a long time. If I add more words (my list is longer), the code fails. Plus, I'm using the token syntax with multiple versions of one word on the token list line. These words are not part of the bit32 list. Can I do this?

**Comment by 3rdIteration** [comment]
Firstly, ElectrumV1 wallets don't follow the BIP39 wordlist.

Also, what do you mean by "it fails", have you managed to identify if there are specific words or part of the syntax that are responsible for breaking it? (eg: works with the example I gave, but if you add X, Y or Z then it breaks...) If you add words in a similar fashion to what you are planning to recover, but ensure that they are all from the ElectrumV1 seed word list, does that work?

**Comment by giantmonster** [comment]
I'm not using the ElectrumV1 wordlist however I did when I created the wallet. I was hoping I could use the first four letters and allow error management to change as needed. This is not a good idea. I'll find the wordlist and adjust manually. I assume it's in the btcrecover directories? 
This is the error I'm currently getting with a non-standard, 41 word long list:
python3 seedrecover.py --dsw --wallet-type electrum1 --addressdb ../_WORDLISTS.txt --tokenlist ../_WORDLISTS.txt --mnemonic-length 12 --language en --addr-limit 5 --no-eta

Starting seedrecover 1.3.0-CryptoGuide, btcrecover 1.3.0-Cryptoguide on Python 3.8.2 64-bit, 21-bit unicodes, 64-bit ints
Loading address database ...
Traceback (most recent call last):
  File "seedrecover.py", line 38, in <module>
    mnemonic_sentence, path_coin = btcrseed.main(sys.argv[1:])
  File "/home/myID/project/btcrecover/btcrecover/btcrseed.py", line 2030, in main
    createdAddressDB = create_from_params["hash160s"] = AddressSet.fromfile(open(args.addressdb, "rb"))
  File "/home/myID/project/btcrecover/btcrecover/addressset.py", line 263, in fromfile
    raise ValueError("unrecognized file format (invalid magic)")
ValueError: unrecognized file format (invalid magic)

**Comment by 3rdIteration** [comment]
The ElectrumV1 wordlist is here: https://github.com/3rdIteration/btcrecover/blob/master/btcrecover/wordlists/electrum1-en.txt

A BIP39 wallet could do something really non-standard and create a seed from any arbitrary list of words, as it is the words themselves that are hashed. (I have never seen a wallet that does this, so it probably isn't what you have done) ElectrumV1 seeds are different in that they basically hash the index number of each word (basically the row number for that word, with the rows starting from 0), so each word in an ElectrumV1 must be part of a valid wordlist. (Though the actual list words could change, the relative position of each word within the list would need to be preserved)

Basically the error you are seeing is because the word you are searching for is not in the ElectrumV1 word list. (And therefore can't be converted from a word to an index)

**Comment by 3rdIteration** [comment]
Reported Bugs resolved - No additional user support questions raised
